R/type-vctr.R

```diff
@@ -49,16 +49,21 @@
 #' @param .data Foundation of class. Must be a vector
 #' @param ... Name-value pairs defining attributes
 #' @param class Name of subclass.
+#' @param inherit_base_type Does this class extend the base type of `.data`?
+#'   i.e. does the resulting object extend the behaviour the underlying
+#'   type?
 #' @export
 #' @keywords internal
 #' @aliases vctr
-new_vctr <- function(.data, ..., class = character()) {
+new_vctr <- function(.data, ..., class = character(), inherit_base_type = TRUE) {
   if (!is_vector(.data)) {
     abort("`.data` must be a vector type.")
   }
 
   nms <- validate_names(.data)
-  attrib <- list(names = nms, ..., class = c(class, "vctrs_vctr"))
+
+  class <- c(class, "vctrs_vctr", if (inherit_base_type) class(.data))
```

Comment:
Would it make sense for it to be `typeof(.data)` so we don't drag all of the `.data` classes along with it? Ideally I guess the caller ensured it was a bare vector with no classes. Also, there is no `typeof()` result for `"data.frame"`. But the others would work.

Reply:
Oh yeah, I forgot that people might be feeding this things other than base classes.

Reply:
This also revealed a subtle bug because `class(1)` is numeric, not double.